opensearch/src/main/java/org/opensearch/sql/opensearch/storage/script/filter/FilterQueryBuilder.java

```diff
@@ -96,16 +100,71 @@ public QueryBuilder visitFunction(FunctionExpression func, Object context) {
         return buildBoolQuery(func, context, BoolQueryBuilder::should);
       case "not":
         return buildBoolQuery(func, context, BoolQueryBuilder::mustNot);
+      case "nested":
+        // TODO Fill in case when adding support for syntax - nested(path, condition)
+        throw new SyntaxCheckException(
+            "Invalid syntax used for nested function in WHERE clause: "
+                + "nested(field | field, path) OPERATOR LITERAL"
+        );
       default: {
         LuceneQuery query = luceneQueries.get(name);
         if (query != null && query.canSupport(func)) {
           return query.build(func);
+        } else if (query != null && query.isNestedFunction(func)) {
+          return handleNestedPredicateExpression(func, query);
```

Comment:
I'm also trying to figure out best way. As I understand, what we want to do is:

```
if isNestedFunc(left) {
  return new NestedQuery(query);
}
```

Is it a little better to move all logic to a `NestedQuery` class?

Reply:
This is close to a solution. We could swap around the arguments in the `ExpressionAnalyzer:visitFunction` to turn this function:
`Function(=) : { nested(...), literal(...)}`
Into this:
`Function(nested) : { function(=) : { nested(...), literal(...)}`

If we do something like this we can move most of the logic to a class extension of `LuceneQuery` and override a `build()` function.  It's a tradeoff as we will need logic to handle only this case in `ExpressionAnalyzer` and the overriding class handling the nested specific logic will need to use the `luceneQueries` map to build the query function. The code isn't very extensible in `pushDownFilter` for changing the left parameter of a predicate expression to something other than a `ReferenceExpression`.

Both implementations will have their drawbacks but without a refactor I can't see a clean implementation for this syntax. At least this isn't a large change that would require much work to refactor.

Reply:
I think I prefer the current solution in this PR more as it may be easier to refactor later if needed. I find swapping the expression tree more clunky with the only benefit being able to move logic to `NestedQuery` class. Do you have an opinion @dai-chen  or any other comments?

Reply:
Sorry for confusion. I'm not suggesting to do this in ExpressionAnalyzer. The pseudocode above is just for this class at this line. I'm just thinking can we move `handleNestedPredicateExpression`, `getNestedPathString`, `buildNested` to a `NestedQuery` class extends `LuceneQuery`?